prov/gni/src/gnix_hashtable.c

```diff
@@ -623,7 +623,14 @@ int _gnix_ht_init(gnix_hashtable_t *ht, gnix_hashtable_attr_t *attr)
 	if (ht->ht_state == GNIX_HT_STATE_READY)
 		return -FI_EINVAL;
 
-	memcpy(&ht->ht_attr, tbl_attr, sizeof(gnix_hashtable_attr_t));
+	ht->ht_attr.ht_initial_size = tbl_attr->ht_initial_size;
+	ht->ht_attr.ht_maximum_size = tbl_attr->ht_maximum_size;
+	ht->ht_attr.ht_increase_step = tbl_attr->ht_increase_step;
+	ht->ht_attr.ht_increase_type = tbl_attr->ht_increase_type;
+	ht->ht_attr.ht_collision_thresh = tbl_attr->ht_collision_thresh;
+	ht->ht_attr.ht_hash_seed = tbl_attr->ht_hash_seed;
+	ht->ht_attr.ht_internal_locking = tbl_attr->ht_internal_locking;
+	ht->ht_attr.destructor = tbl_attr->destructor;
```

Comment:
Yes, however, initializing individual members appears to reduce latency more than the simple structure assignment.
> On May 9, 2016, at 9:45 AM, James Swaro <notifications@github.com> wrote:
> 
> In prov/gni/src/gnix_hashtable.c <https://github.com/ofi-cray/libfabric-cray/pull/812#discussion_r62510144>:
> 
> > @@ -623,7 +623,14 @@ int _gnix_ht_init(gnix_hashtable_t *ht, gnix_hashtable_attr_t *attr)
> >  	if (ht->ht_state == GNIX_HT_STATE_READY)
> >  		return -FI_EINVAL;
> >  
> > -	memcpy(&ht->ht_attr, tbl_attr, sizeof(gnix_hashtable_attr_t));
> > +	ht->ht_attr.ht_initial_size = tbl_attr->ht_initial_size;
> > +	ht->ht_attr.ht_maximum_size = tbl_attr->ht_maximum_size;
> > +	ht->ht_attr.ht_increase_step = tbl_attr->ht_increase_step;
> > +	ht->ht_attr.ht_increase_type = tbl_attr->ht_increase_type;
> > +	ht->ht_attr.ht_collision_thresh = tbl_attr->ht_collision_thresh;
> > +	ht->ht_attr.ht_hash_seed = tbl_attr->ht_hash_seed;
> > +	ht->ht_attr.ht_internal_locking = tbl_attr->ht_internal_locking;
> > +	ht->ht_attr.destructor = tbl_attr->destructor;
> Wouldn't this execute the same code?
> 
> ht->ht_attr = *tbl_attr;
> If not, please explain.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/ofi-cray/libfabric-cray/pull/812/files/abc41c6cb76334b1f6e1fdc1a24af7a70c343be6#r62510144>